device/api/umd/device/cluster.h

```diff
@@ -452,6 +499,11 @@ class tt_device {
     virtual const tt_SocDescriptor& get_soc_descriptor(chip_id_t chip_id) const {
         return soc_descriptor_per_chip.at(chip_id);
     }
+    virtual tt::umd::CoreCoord to(
```

Comment:
rename to something more descriptive (when calling this function, you don't reference return type).
Also since this is in cluster, the "to" might be confusing as it is not referring to "cluster".
Maybe "translate_coord_to"